orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RedBlackStrategy.groovy

```diff
@@ -50,17 +52,32 @@ class RedBlackStrategy implements Strategy, ApplicationContextAware {
 
   @Override
   List<Stage> composeFlow(Stage stage) {
-    def stages = []
-    def stageData = stage.mapTo(StageData)
-    def cleanupConfig = AbstractDeployStrategyStage.CleanupConfig.fromStage(stage)
-
-    Map baseContext = [
-      (cleanupConfig.location.singularType()): cleanupConfig.location.value,
-      cluster                                : cleanupConfig.cluster,
-      credentials                            : cleanupConfig.account,
-      moniker                                : cleanupConfig.moniker,
-      cloudProvider                          : cleanupConfig.cloudProvider,
-    ]
+    List<Stage> stages = []
+    StageData stageData = stage.mapTo(StageData)
+    Map<String, Object> baseContext = AbstractDeployStrategyStage.CleanupConfig.toContext(stageData)
+
+    try {
+      Stage parentCreateServerGroupStage = stage.directAncestors().find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
+
+      if (parentCreateServerGroupStage.status == ExecutionStatus.NOT_STARTED) {
+        // No point in composing the flow if we are called to plan "beforeStages" since we don't have any STAGE_BEFOREs.
```

Comment:
oh no, my crappy bandaid is spreading! :)

Btw, my pr #3133 will obviate the need for this (i will let you know when it's in and can help take advantage of it)

Reply:
Sorry for spreading the crappy bandaid! Very excited for that PR to be merged, I can definitely update this flow once `aroundStages` is officially gone.